libs/designer-ui/src/lib/templates/templateparameters/templateparametersField.tsx

```diff
@@ -0,0 +1,146 @@
+import type { Template } from '@microsoft/logic-apps-shared';
+import type { EventHandler } from '../../eventhandler';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+
+const labelStyles: Partial<ILabelStyles> = {
+  root: {
+    display: 'inline-block',
+    minWidth: '120px',
+    verticalAlign: 'top',
+    padding: '0px',
+  },
+};
+
+const fieldStyles: IStyle = {
+  display: 'inline-block',
+  flexGrow: 1,
+  flexShrink: 1,
+  flexBasis: 'auto',
+};
+
+const textFieldStyles: Partial<ITextFieldStyles> = {
+  root: fieldStyles,
+};
+
+export interface TemplateParameterFieldDetails {
+  name: string;
+  value: string;
+  description?: string;
+  type: string;
+}
+
+const NAME_KEY = 'name';
+const TYPE_KEY = 'type';
+const VALUE_KEY = 'value';
+const DESCRIPTION_KEY = 'description';
+
+export interface TemplateParameterUpdateEvent {
+  name: string;
+  newDefinition: Template.ParameterDefinition;
+}
+
+export type TemplateParameterUpdateHandler = EventHandler<TemplateParameterUpdateEvent>;
+
+export interface TemplateparameterFieldProps {
+  definition: Template.ParameterDefinition;
+  validationError: string | undefined;
+  onChange?: TemplateParameterUpdateHandler;
+}
+
+export const TemplateparameterField = ({ definition, validationError, onChange }: TemplateparameterFieldProps): JSX.Element => {
+  const [value, setValue] = useState<string | undefined>(stringifyValue(definition.value));
+
+  const intl = useIntl();
+
+  const parameterDetails: TemplateParameterFieldDetails = {
+    name: `${definition.name}-${NAME_KEY}`,
+    value: `${definition.name}-${VALUE_KEY}`,
+    description: `${definition.name}-${DESCRIPTION_KEY}`,
+    type: `${definition.name}-${TYPE_KEY}`,
+  };
+
+  const nameTitle = intl.formatMessage({
+    defaultMessage: 'Name',
+    id: 'm8Q61y',
+    description: 'Parameter Field Name Title',
+  });
+  const typeTitle = intl.formatMessage({
+    defaultMessage: 'Type',
+    id: 'tNoZx2',
+    description: 'Parameter Field Type Title',
+  });
+  const descriptionTitle = intl.formatMessage({
+    defaultMessage: 'Description',
+    id: 'UXDOiw',
+    description: 'Parameter Field Description Title',
+  });
+  const valueTitle = intl.formatMessage({
+    defaultMessage: 'Value',
+    id: 'ClZW2r',
+    description: 'Parameter Field Value Title',
+  });
+  const valueDescription = intl.formatMessage({
+    defaultMessage: 'Enter value for parameter.',
+    id: 'rSIBjh',
+    description: 'Parameter Field Value Placeholder Text',
+  });
+
+  const onValueChange = (_event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string): void => {
+    handleValueChange(newValue);
+  };
+
+  const handleValueChange = (value?: string) => {
+    setValue(value);
+
+    onChange?.({
+      name: definition.name,
+      newDefinition: { ...definition, value },
+    });
+  };
+
+  return (
+    <>
+      <div className="msla-workflow-parameter-field">
```

Comment:
NIT: I know this might be just a scratch and also the workflow parameters styles match the expected styles for this interfaces, however it would be great to name them differently or have a common styles name that is not attached to workflow-parameter wording.